core/trino-main/src/main/java/io/trino/memory/ClusterMemoryManager.java

```diff
@@ -317,7 +415,57 @@ private boolean isClusterOutOfMemory()
 
     private QueryMemoryInfo createQueryMemoryInfo(QueryExecution query)
     {
-        return new QueryMemoryInfo(query.getQueryId(), query.getTotalMemoryReservation().toBytes());
+        Multimap<String, TaskMemoryInfo> taskMemoryInfos = ImmutableMultimap.of();
+        if (runsWithTaskLevelRetries(query)) {
+            taskMemoryInfos = createTaskMemoryInfos(query);
+        }
+        return new QueryMemoryInfo(query.getQueryId(), query.getTotalMemoryReservation().toBytes(), taskMemoryInfos);
+    }
+
+    private boolean runsWithTaskLevelRetries(QueryExecution query)
+    {
+        return getRetryPolicy(query.getSession()) == RetryPolicy.TASK;
+    }
+
+    private Multimap<String, TaskMemoryInfo> createTaskMemoryInfos(QueryExecution query)
```

Comment:
might be an artifact from the old implementation (since you already send the memory reservation as part of the memory reservation response)